src/leap/common/tests/test_events.py

```diff
@@ -15,414 +15,156 @@
 # You should have received a copy of the GNU General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-import unittest
-import sets
-import time
-import socket
-import threading
-import random
-
-
-from mock import Mock
-from protobuf.socketrpc import RpcService
-from leap.common import events
-from leap.common.events import (
-    server,
-    client,
-    mac_auth,
-)
-from leap.common.events.events_pb2 import (
-    EventsServerService,
-    EventsServerService_Stub,
-    EventsClientService_Stub,
-    EventResponse,
-    SignalRequest,
-    RegisterRequest,
-    PingRequest,
-    SOLEDAD_CREATING_KEYS,
-    CLIENT_UID,
-)
 
+import os
+import logging
+import time
 
-port = 8090
+from twisted.trial import unittest
+from twisted.internet import defer
 
-received = False
+from leap.common.events import server
+from leap.common.events import client
+from leap.common.events import txclient
+from leap.common.events import catalog
+from leap.common.events.errors import CallbackAlreadyRegisteredError
 
 
-class EventsTestCase(unittest.TestCase):
+if 'DEBUG' in os.environ:
+    logging.basicConfig(level=logging.DEBUG)
 
-    @classmethod
-    def setUpClass(cls):
-        server.EventsServerDaemon.ensure(8090)
-        cls.callbacks = events.client.registered_callbacks
 
-    @classmethod
-    def tearDownClass(cls):
-        # give some time for requests to be processed.
-        time.sleep(1)
+class EventsGenericClientTestCase(object):
 
     def setUp(self):
-        super(EventsTestCase, self).setUp()
+        self._server = server.ensure_server(
+            emit_addr="tcp://127.0.0.1:0",
+            reg_addr="tcp://127.0.0.1:0")
+        self._client.configure_client(
```

Comment:
I don't see where this `self._client` is initializated, it looks like it should be `self._client = client.configure_client(`. But weirdly I get an unrelated error by running the test:
```
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
Selectables:
ZmqPubConnection(ZmqFactory(), [])
ZmqPullConnection(ZmqFactory(), [])
ZmqPullConnection(ZmqFactory(), [])
ZmqPubConnection(ZmqFactory(), [])

leap.common.tests.test_events.EventsClientTestCase.test_client_receives_signal
```